### ISSUE TITLE
Missing proxy setter for WebContext

### ISSUE THREAD

**Issue by jangernert** [gtk-rs/webkit2gtk-rs#81]
[`set_network_proxy_settings`](https://valadoc.org/webkit2gtk-4.0/WebKit.WebContext.set_network_proxy_settings.html) in the `WebContextExt` trait  as well as the `NetworkProxyMode` enum  seem to be missing.  [`NetworkProxySettings`](https://gtk-rs.org/webkit2gtk-rs/webkit2gtk/struct.NetworkProxySettings.html) however exist.

**Comment by antoyo** [comment]
It seems like it was ignored in the [`Gir.toml`](https://github.com/gtk-rs/webkit2gtk-rs/blob/master/Gir.toml#L259) and when `NetworkProxySettings` stopped being ignored, the function didn't.
I guess that would just need this ignore to be removed to make it work.

**Comment by sciguy16** [comment]
Is there anything blocking this being included? Would you like me to open a PR to remove the ignore in Gir.toml and rerun `make gir`?

**Comment by antoyo** [comment]
@sciguy16: I don't know. You can try regenerating the bindings without the ignore to see if it works!

**Comment by sciguy16** [comment]
Generating the bindings seemed to work, it generated these changes:
~~~diff
diff --git a/Gir.toml b/Gir.toml
index 8f04cd1..95ff067 100644
--- a/Gir.toml
+++ b/Gir.toml
@@ -255,9 +255,6 @@ module_name = "uri_response"
 name = "WebKit2.WebContext"
 status = "generate"
 generate_builder = true
-    [[object.function]]
-    name = "set_network_proxy_settings"
-    ignore = true

 [[object]]
 name = "WebKit2.WebsiteDataManager"
diff --git a/src/auto/web_context.rs b/src/auto/web_context.rs
index a425476..f4eb54f 100644
--- a/src/auto/web_context.rs
+++ b/src/auto/web_context.rs
@@ -12,6 +12,12 @@ use crate::FaviconDatabase;
 #[cfg(any(feature = "v2_26", feature = "dox"))]
 #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_26")))]
 use crate::GeolocationManager;
+#[cfg(any(feature = "v2_16", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_16")))]
+use crate::NetworkProxyMode;
+#[cfg(any(feature = "v2_16", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_16")))]
+use crate::NetworkProxySettings;
 use crate::Plugin;
 #[cfg(any(feature = "v2_4", feature = "dox"))]
 #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_4")))]
@@ -325,6 +331,15 @@ pub trait WebContextExt: 'static {
     #[doc(alias = "webkit_web_context_set_favicon_database_directory")]
     fn set_favicon_database_directory(&self, path: Option<&str>);

+    #[cfg(any(feature = "v2_16", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_16")))]
+    #[doc(alias = "webkit_web_context_set_network_proxy_settings")]
+    fn set_network_proxy_settings(
+        &self,
+        proxy_mode: NetworkProxyMode,
+        proxy_settings: Option<&mut NetworkProxySettings>,
+    );
+
     #[doc(alias = "webkit_web_context_set_preferred_languages")]
     fn set_preferred_languages(&self, languages: &[&str]);

@@ -767,6 +782,22 @@ impl<O: IsA<WebContext>> WebContextExt for O {
         }
     }

+    #[cfg(any(feature = "v2_16", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_16")))]
+    fn set_network_proxy_settings(
+        &self,
+        proxy_mode: NetworkProxyMode,
+        proxy_settings: Option<&mut NetworkProxySettings>,
+    ) {
+        unsafe {
+            ffi::webkit_web_context_set_network_proxy_settings(
+                self.as_ref().to_glib_none().0,
+                proxy_mode.to_glib(),
+                proxy_settings.to_glib_none_mut().0,
+            );
+        }
+    }
+
     fn set_preferred_languages(&self, languages: &[&str]) {
         unsafe {
             ffi::webkit_web_context_set_preferred_languages(
~~~

But a `cargo build` failed since the Cairo-rs dependency is broken (pending https://github.com/gtk-rs/webkit2gtk-rs/pull/88)

**Comment by ttys3** [comment]
I have encountered the dependency problem when tried the master branch, That is a nightmare. Everything failed to compile.

So I have to added the feature my self https://github.com/ttys3/webkit2gtk-rs/commit/0d78fb0c6c8c703ab39a96ad6ef6b430b8863f70

And this works good for now (in production).

**Comment by antoyo** [comment]
@sciguy16 : Can you please open a PR with this patch?

**Comment by ttys3** [comment]
> @sciguy16 : Can you please open a PR with this patch?

done, please review

**Comment by antoyo** [comment]
@ttys3 : It seems like it was added manually instead of using the code generator, so I asked [here](https://github.com/gtk-rs/webkit2gtk-rs/pull/91#issuecomment-871358975) to do it since they do the codegen anyway.

**Comment by jangernert** [comment]
Fixed with `0.14`
